src/test/java/com/cloudbees/hudson/plugins/folder/ChildNameGeneratorAltTest.java

```diff
@@ -0,0 +1,549 @@
+                "child_th-b7a6e5662f26eb036090308",
+                "leanbh_c-66fe5ac0be4a896280ef09f",
+                "________-97e4b38574769f9d9968fe9", // ребенок пять
+                "___-d22e9fe51690274d8262bda", // 儿童六
+                "_____7-6d2219439eec0df19863ab8", // 아이 7
+                "nin_o_oc-782e3bad2d233732a03f9dd"
+        );
+        for (String name: Arrays.asList(
+                "child-one",
+                "child_two",
+                "child three",
+                "leanbh cúig",
+                "ребенок пять",
+                "儿童六",
+                "아이 7",
+                "niño ocho"
+        )) {
+            checkChild(instance, name);
+        }
+    }
+
+    private void checkChild(ComputedFolderImpl instance, String idealName) throws IOException {
+        String encodedName = encode(idealName);
+        FreeStyleProject item = instance.getItem(encodedName);
+        assertThat("We have an item for name " + idealName, item, notNullValue());
+        assertThat("The root directory if the item for name " + idealName + " is mangled",
+                item.getRootDir().getName(), is(mangle(idealName)));
+        File nameFile = new File(item.getRootDir(), ChildNameGenerator.CHILD_NAME_FILE);
+        assertThat("We have the " + ChildNameGenerator.CHILD_NAME_FILE + " for the item for name " + idealName,
+                nameFile.isFile(), is(true));
+        String name = FileUtils.readFileToString(nameFile);
+        assertThat("The " + ChildNameGenerator.CHILD_NAME_FILE + " for the item for name " + idealName
+                + " contains the encoded name", name, is(encodedName));
+    }
+
+    public static String encode(String s) {
+        // We force every name to NFD to ensure that the test works irrespective of what the filesystem does
+        return Normalizer.normalize(s, Normalizer.Form.NFD);
+    }
+
+    public static String mangle(String s) {
+        // We force every name to NFD to ensure that the test works irrespective of what the filesystem does
+        s = Normalizer.normalize(s, Normalizer.Form.NFD);
+        String hash = Util.getDigestOf(s);
+        String base = Normalizer.normalize(s, Normalizer.Form.NFD).toLowerCase(Locale.ENGLISH);
+        StringBuilder buf = new StringBuilder(32);
+        for (char c : base.toCharArray()) {
+            if (buf.length() >= 8) break;
+            if (('A' <= c && c <= 'Z')
+                    || ('a' <= c && c <= 'z')
+                    || ('0' <= c && c <= '9')) {
+                buf.append(Character.toLowerCase(c));
+            } else {
+                buf.append('_');
+            }
+        }
+        buf.append('-');
+        buf.append(hash.substring(0, 23));
+        return buf.toString();
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static class ComputedFolderImpl extends ComputedFolder<FreeStyleProject> {
+
+        private Set<String> fatalKids = new TreeSet<String>();
```

Comment:
Probably do not need all this machinery in this test. (Suppose this is a copy-paste?)